R/methods-Nodes.R

```diff
@@ -0,0 +1,40 @@
+# Methods for Node and NodeList objects
+
+
+# Accessors for fanciness
+setGeneric("id", function(object) standardGeneric("id"))
+setGeneric("id<-", function(object, value) standardGeneric("id<-"))
+setGeneric("x", function(object) standardGeneric("x"))
+setGeneric("x<-", function(object, value) standardGeneric("x<-"))
+setGeneric("y", function(object) standardGeneric("y"))
+setGeneric("y<-", function(object, value) standardGeneric("y<-"))
+
+#' @include methods-Links.R
+## Methods for Nodes
+setMethod("id", "Node", function(object) id(object@id))
+setMethod("x", "Node", function(object) object@x)
+setMethod("y", "Node", function(object) object@y)
+setMethod("color", "Node", function(object) object@color)
+setMethod("weight", "Node", function(object) object@weight)
+setMethod("id<-", "Node", function(object, value) {object@NodeId@id <- value; validObject(object); object})
+setMethod("x<-", "Node", function(object, value) {object@x <- value; validObject(object); object})
+setMethod("y<-", "Node", function(object, value) {object@y <- value; validObject(object); object})
+setMethod("color<-", "Node", function(object, value) {object@color <- value; validObject(object); object})
+setMethod("weight<-", "Node", function(object, value) {object@weight <- value; validObject(object); object})
+
+## Methods for NodeId
+setMethod("id", "NodeId", function(object) object@value)
```

Comment:
should this method be "value", since the slot is named "value"?

Reply:
maybe.. but value is a bit vague really. if i have a thing in hand and dont remember whether its a Node or NodeId, it might be nice to be able to use `id` in either case, is what i figured. i dont mind aliasing though for cases like these.